sheep_cli/src/main.rs

```diff
@@ -208,11 +211,37 @@ fn load_images(input: &[String]) -> Vec<InputSprite> {
         .collect()
 }
 
-fn write_files<S: Serialize>(output_path: &str, outbuf: RgbaImage, meta: S, pretty: bool) {
-    outbuf
-        .save(format!("{}.png", output_path))
-        .expect("Failed to save image");
+fn write_image(output_path: &str, sheet: &SpriteSheet, compress: bool) {
+    let filename = format!("{}.png", output_path);
+
+    if compress {
+        let mut png_bytes = Vec::<u8>::new();
+        {
+            let mut encoder =
+                png::Encoder::new(&mut png_bytes, sheet.dimensions.0, sheet.dimensions.1);
+            encoder.set_color(png::ColorType::RGBA);
+            encoder.set_depth(png::BitDepth::Eight);
+            encoder.set_compression(png::Compression::Fast);
+            encoder.set_filter(png::FilterType::NoFilter);
+            let mut writer = encoder.write_header().unwrap();
+            writer.write_image_data(&sheet.bytes).unwrap();
+        }
+
+        let options = oxipng::Options::from_preset(3);
+        let compressed = oxipng::optimize_from_memory(png_bytes.as_slice(), &options)
+            .expect("Failed to compress png");
+        let mut file = File::create(filename).expect("Failed to create image file");
+        file.write_all(compressed.as_slice())
+            .expect("Failed to save image");
+    } else {
+        RgbaImage::from_vec(sheet.dimensions.0, sheet.dimensions.1, sheet.bytes.clone())
```

Comment:
Honestly, I'd be in favor of dropping this and just using the encoder from the `png` crate for both branches. Then you can do:

```rust
let mut png_bytes = // ...

if compress {
    let options = oxipng::Options::from_preset(3);
    png_bytes = oxipng_optimize_from_memory(png_bytes.as_slice(), &options);
    // or make a temporary var and assign it if the borrow checker fights this somehow
}

// write bytes
```

This way it's easier to add other options later.